index.js

```diff
@@ -1,38 +1,44 @@
-/* jshint node: true */
 'use strict';
 
-var derequire = require('broccoli-derequire');
+const Derequire = require('broccoli-derequire');
+const BroccoliDebug = require('broccoli-debug');
 
 module.exports = {
-  name: 'ember-derequire',
+  name: require('./package').name,
 
-  included: function(app) {
+  included(app) {
     this._super.included.apply(this, arguments);
     this.hostBuildOptions = app.options.derequire || {};
 
-    var defaultOptions = {enabled: this.app.env !== 'test'};
+    let defaultOptions = { enabled: this.app.env !== 'test' };
```

Comment:
Hi @simonihmig, I'm curious if the default condition here is the root cause of the test issue you noticed. I wonder if setting it to false would make any difference `{ enabled: false }`

Reply:
Hm, I don't think it's the root issue, rather this "workaround" would obfuscate the issue. 

So the root issue is still that the `derequire` library would only do its work, when it sees `define` and `require` as variables declared in the current scope, and won't do anything when they are used as globals. At least that's my understanding. And that's why there were these `var define = define; var require = require;` injected. 

Whether that approach makes that much sense is another question, at least in our case. After all `derequire` could easily also transform those function calls even when they refer to globals. I guess this was some safety concern, which we work around by adding those variable declarations... 😕

When we disable the addon by default - or rather we would probably want to *enable* it only in a production build things would work fine in dev. But that means you are developing and testing things differently than what you deploy to users, leaving open the danger of breaking the app in production while everything "works on my machine"(TM)

I *think* for our use case it would actually be more reasonable if we could make `derequire` ignore its current policy and just transform everything that is called `define()` or `require()`. Then we would omit that rather silly thing to add those local declarations just to make `derequire` happy. After all all current Ember code expects those global functions to exist anyway. But that would require upstream changes.

I can create an issue to bring this topic up, if we agree that is the path forward?

Reply:
Yes, that makes sense. Thank you :)